shared/ClrFacade/Internal.cs

```diff
@@ -146,12 +161,22 @@ public static Type GetType(string typename)
 
    public static int CurrentObject(IntPtr returnValue)
    {
-      var result = DataConverter?.CurrentObject;
-
-      var tempRetVal = RSharpGenericValueExtensions.FromObject(result);
-
-      Marshal.StructureToPtr(tempRetVal, returnValue, false);
-      return 1;
+      try
+      {
+         var result = DataConverter?.CurrentObject;
+         var tempRetVal = RSharpGenericValueExtensions.FromObject(result);
+         Marshal.StructureToPtr(tempRetVal, returnValue, false);
+      }
+      catch (Exception ex)
+      {
+         if (!logThroughR(ex))
```

Comment:
Once you have a refactor on the ```logThroughR``` method done, can you pull all this into a single common method. We are always doing the same thing, so it should be only written once and used from everywhere.

Reply:
The common code that can be put into a method is this

```
       if (!logThroughR(ex))
       {
          var tempRetVal = RSharpGenericValueExtensions.FromObject(LastCallException);
          Marshal.StructureToPtr(tempRetVal, returnValue, false);
          return -1;
       }
```